### PR TITLE
chore: DAH-2296 refactor and test leasing agent support

### PR DESCRIPTION
**Pull request by alulabeshue-sfgov**
[DAH-2296](https://sfgovdt.jira.com/browse/DAH-2296)

[DAH-2296]: https://sfgovdt.jira.com/browse/DAH-2296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/api/v1/listing_interest_controller.rb [hunk 1]
```diff
@@ -24,17 +25,17 @@ def confirmation
       end
 
       listing_id = listing_id_map(b)
```
Comment (tallulahkay) [inline app/controllers/api/v1/listing_interest_controller.rb:27]: could you add comments here explaining what r, x, and e mean?

File: config/routes.rb [hunk 2]
```diff
@@ -37,7 +37,7 @@ def matches?(_)
           get 'eligibility' => 'listings#eligibility'
         end
       end
-      get 'trk' => 'email#confirmation'
+      get 'trk' => 'listing_interest#index'
```
Comment (tallulahkay) [inline config/routes.rb:40]: I actually like confirmation better - I think it's clearer

File: config/routes.rb [hunk 3]
```diff
@@ -37,7 +37,7 @@ def matches?(_)
           get 'eligibility' => 'listings#eligibility'
         end
       end
-      get 'trk' => 'email#confirmation'
```
Comment (tallulahkay) [inline config/routes.rb:40]: what is trk?

File: app/controllers/api/v1/listing_interest_controller.rb [hunk 4]
```diff
@@ -24,17 +25,17 @@ def confirmation
...
-      redirect_to "/confirming_email?listing=#{listing_id}&response=#{r}"
+      redirect_to "/listing_interest?listing=#{listing_id}&response=#{r}"
```
Comment (tallulahkay) [inline app/controllers/api/v1/listing_interest_controller.rb:28]: can you also explain b, a, r, s?

**Review by tallulahkay — COMMENTED** [review]
This looks good - just wondering if there's a way to avoid hard coding the listings? can we generate a listingIdToNameMap by calling an endpoint?

File: config/routes.rb (hunk 3 above)
Comment (alulabeshue-sfgov) [reply config/routes.rb:40]: It is short for tracking. Email templates in salesforce campaign monitor have a size limit for links and the token itself was getting long so we decided to use shorter names in the api like `trk` for tracking and the param `t` for token.

File: config/routes.rb (hunk 2 above)
Comment (alulabeshue-sfgov) [reply config/routes.rb:40]: Ok. I was thinking it might be confused with the email confirmations we have today for when a user signs up for an account.

Comment (tallulahkay) [reply config/routes.rb:40]: listing_interest#confirmation?

**Review by tallulahkay — APPROVED** [review]
Love the comments!

**Pull request merged**